In [1]:
import pandas as pd

## Language Analysis of active SE purposes repositories:

In [2]:
df_SE = pd.read_csv("data_only_notebooks/language_info_active_SE_purpose_repos.csv")
print(f"Total number of notebooks belonging to active SE purposes repositories: {len(df_SE)}")

Total number of notebooks belonging to active SE purposes repositories: 8631


Number of distinct languages of notebooks in the SE split:

In [3]:
df_SE['language'].nunique()

39

In [4]:
for key,item in df_SE['language'].value_counts().items():
    print(key, item)
    print("\n")

python 5809


python3 1754


Python 655


Groovy 98


markdown 41


unknown 33


Scala 31


julia 30


python2 25


powershell 24


sql 18


Prolog 14


qsharp 12


R 10


SQL 9


Batchfile 8


pyspark3kernel 8


Java 7


Kotlin 7


Clojure 7


Java11 5


Julia 4


PowerShell 3


scala 2


Dockerfile 2


pyspark 2


c 1


dyalog-kernel 1


aa 1


lua 1


conda-env-DeepPurpose-py 1


Matlab 1


Markdown 1


julia-1.7 1


pycharm-7906112d 1


Lua 1


Lisp 1


JavaScript 1


nlpaug_dev 1




In [10]:
with open("language_only_nbs_SE_repos.txt", 'w', encoding='utf-8') as f:
    for key, value in (df_SE['language'].value_counts()).items():
        print(key, value, file=f)

## Jupyter notebook count per language in the SE purpose split:

Python: 8243 (python 5809 + python3 1754 + Python 655 + python2 25)

Groovy 98

markdown 41

Julia: 34 (Julia 4 + julia 30)

unknown 33

Scala 33 (Scala 31 + scala 2

powershell 24

sql 27 (18 +9)

Prolog 14

qsharp 12

Java: 12 (java 7 + Java11 5)

R 10

Batchfile 8

Pyspark: 10 (pyspark3kernel 8 + pyspark 2)

Kotlin 7

Clojure 7

## Language Analysis of active non SE purposes repositories:

In [5]:
df_non_SE = pd.read_csv("data_only_notebooks/language_info_active_non_SE_purpose_repos.csv")
print(f"Total number of notebooks belonging to active non SE purposes repositories: {len(df_non_SE)}")

Total number of notebooks belonging to active non SE purposes repositories: 27837


Number of distinct languages of notebooks in the non SE split:

In [6]:
df_non_SE['language'].nunique()

40

In [7]:
for key,item in df_non_SE['language'].value_counts().items():
    print(key, item)
    print("\n")

python 23737


python3 1852


Python 1003


unknown 340


julia 279


matlab 123


R 101


markdown 96


qsharp 61


python2 38


Java 34


swift 27


pyspark 18


Lua 18


scala 16


Julia 16


INI 12


Ruby 11


Shell 7


Kotlin 6


Markdown 6


Batchfile 6


scala211 4


conda_python3 3


Matlab 3


JavaScript 3


ir 2


bash 2


conda_pytorch_p36 2


Rust 1


haskell 1


TeX 1


SQL 1


XML 1


Haskell 1


sql 1


octave 1


c++ 1


CoffeeScript 1


lua 1




In [9]:
with open("language_only_nbs_non_SE_repos.txt", 'w', encoding='utf-8') as f:
    for key, value in (df_non_SE['language'].value_counts()).items():
        print(key, value, file=f)

## Jupyter notebook count per language in the non SE purpose split:

Python: 26,633 (python 23737 + python3 1852 + Python 1003 +python2 38 + conda_python3 3)

Misc: 340 (unknown 340)

Julia: 295 (julia 279 + Julia 16)

Matlab: 126 (matlab 123 + Matlab 3)

Markdown: 102 (markdown 96 + Markdown 6)

R: 101

Qsharp: 61

Java: 34

Swift: 27

Scala: 20 (scala 16 + scala211 4)

Pyspark: 18

Lua: 18

INI: 12

Ruby: 11

Shell: 7

Kotlin: 6

Batchfile: 6
